app/sections/collection-header.tsx

```diff
@@ -0,0 +1,142 @@
+import type {
+    HydrogenComponentProps,
+    HydrogenComponentSchema,
+} from '@weaverse/hydrogen';
+import { forwardRef } from 'react';
+import { useLoaderData } from '@remix-run/react';
+import { CollectionDetailsQuery } from 'storefrontapi.generated';
+
+interface HeaderProps extends HydrogenComponentProps {
+    heightSectionDesktop: number;
+    heightSectionMobile: number;
+    enableBackground: boolean;
+    textPosition: string;
+    overlayBackground: number;
+    enableOverlay: boolean;
+}
+
+let CollectHeader = forwardRef<HTMLElement, HeaderProps>((props, ref) => {
+    let { heightSectionDesktop, heightSectionMobile, enableBackground, textPosition, overlayBackground, enableOverlay, ...rest } = props;
+    let { collection } = useLoaderData<
+        CollectionDetailsQuery & {
+            collections: Array<{ handle: string; title: string }>;
+        }
+    >();
+    let backgroundStyle = {};
+    if (enableBackground) {
+        backgroundStyle = {
+            backgroundImage: `url('${collection?.image?.url}')`,
+            '--header-height-desktop': `${heightSectionDesktop}px`,
+            '--header-height-mobile': `${heightSectionMobile}px`
+        };
+    }
+    let textStyle = {};
+    if (enableBackground) {
+        textStyle = {
+            color: '#fff',
+        };
+    } else {
+        textStyle = {
+            color: '#333',
+            position: 'relative',
+        };
+    }
+    let overlayStyle = {};
+    if (enableOverlay && enableBackground) {
+        overlayStyle = {
+            backgroundColor: `rgba(0, 0, 0, ${overlayBackground})`,
```

Comment:
Set the `overlayOpacity` to a css variable, then get it using `var()` in a class name like what you did with section height, don't inline style this